testing/asserts.ts

```diff
@@ -864,3 +866,58 @@ export function unimplemented(msg?: string): never {
 export function unreachable(): never {
   throw new AssertionError("unreachable");
 }
+
+let snapshotFile: Record<string, string> | undefined = undefined;
+const updatedSnapshotFile: Record<string, unknown> = {};
+const snapshotMap: Record<string, number> = {};
+
+export async function assertSnapshot(context: Deno.TestContext, actual: unknown) {
+  const name = getName(context);
+  const count = getCount();
+  const testName = `${name} #${count}`;
+  const isUpdate = Deno.args.includes('--update');
+  if (!snapshotFile) {
+    const snapshotPath = getSnapshotPath();
+    await ensureFile(snapshotPath);
+    const {snapshot} = await import(snapshotPath);
+    snapshotFile = snapshot;
+  }
+  const snapshot = snapshotFile?.[testName] ?? '';
+  if (isUpdate) {
+    let isEqual = true;
+    try {
+      assertEquals(_format(actual), snapshot.slice(1, -1));
```

Comment:
Given these are reused in both the `if` and `else` blocks, can we extract these out above the `if` statement for better readability/maintainability e.g.

```ts
// --snip--
const _actual = _format(actual);
const _expected = snapshot.slice(1, -1);
if (isUpdate) {
 // --snip--
```

Reply:
Done